faf-commons-data/src/main/java/com/faforever/commons/replay/ReplayDataParser.java

```diff
@@ -34,30 +35,47 @@
   private static final int LUA_TABLE_START = 4;
   private static final int LUA_TABLE_END = 5;
   private final Path path;
+  private final ObjectMapper objectMapper;
+  @Getter
+  private ReplayMetadata metadata;
+  @Getter
+  private String replayPatchFieldId;
+  @Getter
+  private byte[] data;
+  @Getter
+  private String map;
+  @Getter
+  private Map<String, Map<?, ?>> mods;
+  @Getter
   private Map<Integer, Map<String, Object>> armies;
   private int randomSeed;
+  @Getter
   private List<ChatMessage> chatMessages;
+  @Getter
   private Map<Integer, Map<Integer, AtomicInteger>> commandsPerMinuteByPlayer;
   private float x;
   private float y;
   private float z;
   private float w;
   private float scale;
   private int ticks;
+  @Getter
   private List<GameOption> gameOptions;
 
-  public ReplayDataParser(Path path) {
+  public ReplayDataParser(Path path, ObjectMapper objectMapper) {
     this.path = path;
+    this.objectMapper = objectMapper;
     armies = new HashMap<>();
     chatMessages = new ArrayList<>();
     commandsPerMinuteByPlayer = new HashMap<>();
+    parse();
   }
 
   @VisibleForTesting
   String readString(LittleEndianDataInputStream dataStream) throws IOException {
     ByteArrayOutputStream out = new ByteArrayOutputStream();
     byte tempByte;
-    while ((tempByte =  dataStream.readByte()) != 0) {
+    while ((tempByte = dataStream.readByte()) != 0) {
       out.write(tempByte);
     }
```

Comment:
Can we use IOUtils.copy here?

Reply:
Think so

Reply:
Hmm seems like you can't because of the little endian stream. at least the test fails